### PR TITLE
Use JvmTestSuite + JacocoReport

### LINKED ISSUE #5991: Replace custom jacoco report aggregation with Gradle's one
See https://docs.gradle.org/7.4/samples/sample_jvm_multi_project_with_code_coverage_distribution.html and https://docs.gradle.org/7.4/userguide/jacoco_report_aggregation_plugin.html

Hints:
* Replace usage of `org.projectnessie.buildsupport.jacoco` plugin in `nessie-conventions.gradle.kts` with `jacoco`
* Update `:nessie-code-coverage` project, remove `org.projectnessie.buildsupport.jacoco-aggregator` plugin
* Apply similar changes to projectnessie/cel-java
* Remove the jacoco plugins from projectnessie/gradle-build-plugins

### PR DESCRIPTION
**Pull request by snazy**
Use Gradle's test-suites instead of the custom approach, adopt code coverage as well.

This brings a couple of improvements:
* Class paths for `test` and `intTest` are separated
* When running tests in IntelliJ, it will no longer as for the test-task to use
* Test suites enables the use of the built-in Jacoco report aggregation (see below)
* Can add more test suites later, if necessary - e.g. further split into different use cases and such

Notes about the test-suite changes:
* `test` stays as it is
* `intTest` gets its own source directory, and its own configurations for dependencies
* both `test` and `intTest` inherit common test code and dependencies from `testFixtures`
    
The `code-coverage` aggregator is gone. It was already broken, because it blindly aggregated all projects, even those that mix up different versions of dependencies (think: Scala, Spark, etc), which is incorrect. Gradle's native jacoco-aggregation (the `testCodeCoverageReport` + `intTestCodeCoverageReport` inplicitly tasks introduced w/ this change) properly inherit the dependencies. The correct way to get an aggregated test-coverage is to run (one of) the `*CodeCoverageReport` tasks on the requested server/cli/application project - for example `:nessie-quarkus:testCodeCoverageReport`.

Other Gradle task changes:
* Since there are now two more new source sets (`intTest` + `testFixtures`), there are also more checkstyle tasks. An aggregating task `checkstyle` has been added for each Gradle project.
* Also related to the added source sets: CI workflows have been changed to use the (built in) `assemble` task instead of `jar testClasses` - note: `assemble` includes `javadoc`
* CI workflow test + int-test invocations have been "enhanced" with their corresponding code-coverage aggregation tasks

The PR contains two commits:
1. changes to the common build logic
2. "boring" changes (moving files, dependency configuration adjustments)

Fixes #5991

### CODE REVIEW

**Review by adutra — COMMENTED** [review]
Just a first small round of review, will continue tomorrow.

File: README.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ To deploy the ui (from `ui` directory):
 
 ### Docker image
 
-When running `./gradlew :nessie-quarkus:quarkusBuild -Pdocker -Pnative` a docker image will
+When running `./gradlew :nessie-quarkus:quarkusBuild -Pnative` a docker image will
```
[the remark this answers is not in the document]
Comment (snazy) [reply README.md:135]: see `tools/dockerbuild/README.md` - Quarkus doesn't build the Docker image anymore

File: .github/workflows/pull-request.yml [hunk 2]
```diff
@@ -77,10 +77,7 @@ jobs:
     - name: Gradle / assemble + publish local
       uses: gradle/gradle-build-action@v2
       with:
-        arguments: |
-          assemble
-          publishToMavenLocal
-          --scan
+        arguments: assemble publishToMavenLocal -Puber-jar --scan
```
[the remark this answers is not in the document]
Comment (snazy) [reply .github/workflows/pull-request.yml:80]: not needed (no idea why I added it)

[the remark this answers is not in the document]
Comment (snazy) [reply .github/workflows/pull-request.yml]: removed

File: README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (adutra) [reply README.md:135]: In this case, the whole paragraph is outdated. I can provide a small PR to fix this later.

File: testing/s3minio/build.gradle.kts [hunk 3]
```diff
@@ -41,11 +41,16 @@ dependencies {
 
   compileOnly(libs.errorprone.annotations)
 
-  testImplementation(libs.bundles.junit.testing)
-  testRuntimeOnly(libs.junit.jupiter.engine)
-  testImplementation(libs.hadoop.common) { withSparkExcludes() }
+  // TODO issue in JvmSuitePlugin, that these dependencies are _not_ inherited by default
```
Comment (adutra) [inline testing/s3minio/build.gradle.kts:44]: FYI I could not reproduce the problem. If I remove lines 45-47, I am still able to run all the integration tests in this project and in all dependent projects, e.g. nessie-gc-iceberg-files.

File: gc/gc-iceberg-files/build.gradle.kts [hunk 4]
```diff
@@ -53,28 +53,26 @@ dependencies {
...
 
-  testImplementation(nessieProject("nessie-gc-base-tests"))
-  testImplementation(nessieProject("nessie-s3mock"))
-  testImplementation(nessieProject("nessie-s3minio"))
+  testFixturesApi(nessieProject("nessie-gc-base-tests"))
+  testFixturesApi(nessieProject("nessie-s3mock"))
+  testFixturesApi(nessieProject("nessie-s3minio"))
 
-  testRuntimeOnly(libs.logback.classic)
+  testFixturesApi(libs.logback.classic)
```
[the remark this answers is not in the document]
Comment (adutra) [reply gc/gc-iceberg-files/build.gradle.kts:60]: Applying my own findings to this build script, I was able to narrow the scope of many deps:

```
  testFixturesRuntimeOnly(libs.logback.classic)

  testFixturesCompileOnly(platform(libs.jackson.bom))
  testFixturesAnnotationProcessor(libs.jackson.annotations)

  testFixturesCompileOnly(libs.microprofile.openapi)

  testFixturesApi(platform(libs.awssdk.bom))
  testFixturesApi(libs.awssdk.s3)
  testFixturesImplementation(libs.awssdk.sts)
  testFixturesImplementation(libs.hadoop.aws)

  testFixturesApi(platform(libs.junit.bom))
  testFixturesApi(libs.bundles.junit.testing)
  testFixturesRuntimeOnly(libs.junit.jupiter.engine)
```

Not sure if it's worth double-checking other build scripts, or if we can go ahead and consider the current state as "good enough" :-)

**Review by adutra — CHANGES_REQUESTED** [review]
Impressive work.

1) The code coverage part is working as expected. I was able to generate distinct reports for tests and intTests:

image

2) The test suites are working as expected.

3) The introduction of the test-fixtures plugin is working correctly, although I have a few doubts regarding the scope of some dependencies: most of them seem to be declared as `testFixturesApi` for convenience, but could maybe be included in a more restrictive scope.

File: clients/client/build.gradle.kts [hunk 5]
```diff
@@ -49,14 +49,13 @@ dependencies {
   compileOnly(libs.immutables.value.annotations)
   annotationProcessor(libs.immutables.value.processor)
 
-  testImplementation(libs.guava)
-  testImplementation(libs.bouncycastle.bcprov)
-  testImplementation(libs.bouncycastle.bcpkix)
-  testImplementation(libs.mockito.core)
+  testFixturesApi(libs.guava)
```
[the remark this answers is not in the document]
Comment (snazy) [reply clients/client/build.gradle.kts:52]: Yup, the dependency's inherited by all test suites.

File: servers/quarkus-cli/build.gradle.kts [hunk 6]
```diff
@@ -81,18 +81,21 @@ dependencies {
...
-  testImplementation(project(":nessie-versioned-tests"))
-  testImplementation(project(":nessie-versioned-storage-mongodb"))
-  testImplementation(project(":nessie-versioned-storage-testextension"))
-  testImplementation("io.quarkus:quarkus-jacoco")
-  testImplementation("io.quarkus:quarkus-junit5")
-  testCompileOnly(libs.microprofile.openapi)
-
-  testImplementation(platform(libs.junit.bom))
-  testImplementation(libs.bundles.junit.testing)
+  testFixturesApi(project(":nessie-quarkus-tests"))
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/quarkus-cli/build.gradle.kts:84]: correct

File: servers/quarkus-server/src/test/java/org/projectnessie/server/error/ErrorTestService.java [hunk 7]
```diff
@@ -44,7 +44,7 @@
 import org.projectnessie.versioned.persist.adapter.DatabaseAdapter;
 import org.projectnessie.versioned.persist.store.PersistVersionStore;
 
-/** REST service used to generate a bunch of violations for {@link TestNessieError}. */
+/** REST service used to generate a bunch of violations for {@code TestNessieError}. */
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/quarkus-server/src/test/java/org/projectnessie/server/error/ErrorTestService.java:47]: ah, I moved that class more than once

File: testing/s3minio/build.gradle.kts (hunk 3 above)
Comment (snazy) [reply testing/s3minio/build.gradle.kts:44]: yup - correct. probably a left-over

File: gc/gc-iceberg-files/build.gradle.kts [hunk 8]
```diff
@@ -53,28 +53,24 @@ dependencies {
...
 
-  testImplementation(nessieProject("nessie-gc-base-tests"))
-  testImplementation(nessieProject("nessie-s3mock"))
-  testImplementation(nessieProject("nessie-s3minio"))
+  testFixturesApi(nessieProject("nessie-gc-base-tests"))
+  testFixturesApi(nessieProject("nessie-s3mock"))
+  testFixturesApi(nessieProject("nessie-s3minio"))
 
-  testRuntimeOnly(libs.logback.classic)
+  testFixturesCompileOnly(libs.logback.classic)
```
Comment (adutra) [inline gc/gc-iceberg-files/build.gradle.kts:60]:
```suggestion
  testFixturesRuntimeOnly(libs.logback.classic)
```

Comment (adutra) [reply gc/gc-iceberg-files/build.gradle.kts:60]: Symptom is:

```
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
```

**Review by adutra — APPROVED** [review]
🎉

**Pull request merged**